### ISSUE TITLE
Bug in FullyConnected???

### ISSUE THREAD

**Issue by xuzhm** [apache/incubator-mxnet#8646]
I   built  a simple  network  like this:

```python
    data = mx.symbol.Variable(name="data")
    net = mx.symbol.Pooling(data=data, global_pool=True, kernel=(2, 2), pool_type='avg', name=('stn_pool2'))
    net = mx.symbol.Flatten(data=net, name='flatten')

    w1 = mx.symbol.Variable(name="fc_loc1_weight", init=mx.init.Zero(), lr_mult=0, wd_mult=0)
    b1 = mx.symbol.Variable(name="fc_loc1_bias", init=mx.init.One(), lr_mult=0, wd_mult=0)
    loc1 = mx.symbol.FullyConnected(data=net, num_hidden=6, name="fc_loc1", weight=w1, bias=b1)

    w2 = mx.symbol.Variable(name="fc_loc2_weight", init=mx.init.Zero(), lr_mult=0, wd_mult=0)
    b2 = mx.symbol.Variable(name="fc_loc2_bias", init=mx.init.Zero(), lr_mult=0, wd_mult=0)
    loc2 = mx.symbol.FullyConnected(data=net, num_hidden=6, name="fc_loc2", weight=w2, bias=b2)

    w3 = mx.symbol.Variable(name="fc_loc3_weight", init=mx.init.Zero(), lr_mult=0, wd_mult=0)
    b3 = mx.symbol.Variable(name="fc_loc3_bias", init=mx.init.One(), lr_mult=0, wd_mult=0)
    loc3 = mx.symbol.FullyConnected(data=net, num_hidden=6, name="fc_loc3", weight=w3, bias=b3)

    locsum = mx.symbol.ElementWiseSum(*[loc1, loc2, loc3], name='loc_sum')

    fc       = mx.symbol.FullyConnected(data=locsum, num_hidden=num_classes, name='fc')
    softmax   = mx.symbol.SoftmaxOutput( data=fc,  name='softmax')



then
  
    model = mx.mod.Module(
        context       = devs,
        symbol        = new_sym
    )

   ...

   def norm_stat(d):
        return d
   mon = mx.mon.Monitor(1, norm_stat, pattern='.*')

   ...

 model.fit(train,
              begin_epoch=epoch,
              num_epoch=num_epoch,
              eval_data=val,
              eval_metric=eval_metric,
              kvstore=kv,
              optimizer='sgd',
              optimizer_params=optimizer_params,
              arg_params=arg_params,
              aux_params=aux_params,
    #          initializer=initializer,
              allow_missing=True, # for new fc layer
              #allow_missing=False, # for new fc layer
              monitor = mon,
              batch_end_callback=mx.callback.Speedometer(args.batch_size, 1),
              epoch_end_callback=checkpoint)
```

and  plot  the network:
![image](https://user-images.githubusercontent.com/6814767/32766517-f640117c-c8d4-11e7-9451-3b1206e419a3.png)

but  the monitor  shows:

![image](https://user-images.githubusercontent.com/6814767/32766731-d13cb46a-c8d5-11e7-9028-ae5878a6f1f1.png)

fc_loc1_output  is not same with  fc_loc3_output  , what's wrong?        loc_sum_0  != ( fc_loc1_output + fc_loc2_output + fc_loc3_output  )   ?

@piiswrong

**Comment by szha** [comment]
Thanks for reporting this. Do you mind making the code snippet runnable to ease the efforts on reproducing it? (assuming you already have runnable example). Using dummy data (e.g. `mx.nd.random.uniform`) should suffice.

**Comment by anxiaoxi45** [comment]
[test_fc.py.zip](https://github.com/apache/incubator-mxnet/files/1473785/test_fc.py.zip)
@szha    Here is  my test scripts. 
 The fc output of the monitor in the mx.mod.Module.fit    is very strange  ...

**Comment by anirudh2290** [comment]
@eric-haibin-lin Please tag with "Bug" and "Need triage"

**Comment by szha** [comment]
@anirudh2290 @eric-haibin-lin please refrain from manually adding "Need Triage" label while triaging. Thanks.

**Comment by vrakesh** [comment]
@szha Has this issue been resolved? Requesting an update on the same

**Comment by szha** [comment]
@vrakesh thanks for pinging. I haven't worked on this so not sure what the status is. I'd recommend trying to reproduce the issue with the above code and see if the issue persists.

**Comment by vandanavk** [comment]
@mxnet-label-bot add [Operator]

**Comment by sandeep-krishnamurthy** [comment]
Hello @xuzhm 
Sorry for the delay in our response.
I ran a sample to verify the above behavior
```
data_0 -> fc_0  \
data_1 -> fc_1   \ 
data_2 -> fc_2      => sum
data_3 -> fc_3  /
data_4 -> fc_4 /
```
Then compare the below:
1. End to end in single module
2. Individually calculate output of fc_0, ... fc_4 and then sum it to compare with above result

I find the results are same. And, I am using latest MXNet (1.5.0) got with `pip install mxnet-mkl --pre`

**Comment by lanking520** [comment]
@xuzhm close this issue for now since lacking response. Please feel free to reopen if the above solution doesn't help
